python/ray/data/dataset_pipeline.py

```diff
@@ -573,7 +573,7 @@ def show_windows(self, limit_per_dataset: int = 10) -> None:
             print("=== Window {} ===".format(i))
             ds.show(limit_per_dataset)
 
-    def iter_epochs(self) -> Iterator["DatasetPipeline[T]"]:
+    def iter_epochs(self, max_epoch: int = -1) -> Iterator["DatasetPipeline[T]"]:
```

Comment:
Since users already have a max_epoch in mind, can they just ds.repeat(max_epoch) when creating dataset pipeline?

Reply:
They cannot, since in AIR the system does the repeat.